### PR TITLE
tests: improve image_prefetcher_await_set()

### PR DESCRIPTION
**Pull request by porridge**
### Description

Awaiting prefetcher was [unintentionally always passing](https://redhat-internal.slack.com/archives/C0321S70YK1/p1731415449297889?thread_ts=1731355279.014839&cid=C0321S70YK1) because of running inside an `if` condition.

This change makes it intentional for the:
- potentially-flaky parts to not fail the job, and
- the parts which should in theory never fail - to fail the job

It also adds some messages and a summary file to help investigate failures.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [x] modified existing tests

#### How I validated my change

- created a dummy PR based on this one which forces the pull timeout to 1s to make the prefetching fail: https://github.com/stackrox/stackrox/pull/13514
- looked at one of the jobs
  image
- made sure the query works as expected:
  image

### CODE REVIEW

File: scripts/ci/lib.sh [hunk 1]
```diff
@@ -719,10 +719,59 @@ _image_prefetcher_system_await() {
...
+                </style>
+                </head>
+                <body>
+
+                Waiting for image prefetcher set ${name} to complete timed out.<br>
+                To investigate closer, go to <a target="_blank" href="https://console.cloud.google.com/bigquery">BigQuery</a> and run a query such as the following:
+                <br>
+                <pre>
+EOM
+            cat >> "${prefetcher_help}" "${query}"
```
Comment (mtodor) [inline scripts/ci/lib.sh:763]: nitpick: I like your effort and constructing nice HTML with query wrapped in `pre` for easy copy/paste - but with that, we now have two places that should be adjusted.

I was thinking if we could simplify it. Maybe output all log messages into a file and then output it with `cat` and wrap it in HTML (maybe everything in `pre`). In that case, we have one place where the message is constructed. Or the other way around - create an HTML file and then strip tags and output it to the console (if we have `pandoc` in the container, that would be easy).

**Review by mtodor — APPROVED** [review]
It looks good. Added a few thoughts.

File: scripts/ci/lib.sh (hunk 1 above)
Comment (porridge) [reply scripts/ci/lib.sh:763]: Ideally [spyglass](https://docs.prow.k8s.io/docs/spyglass/) would have a "markdown" lens in addition to the `html` one 🤔 
Actually let me take a stab at writing such lens in the background.

File: scripts/ci/lib.sh [hunk 2]
```diff
@@ -740,18 +789,29 @@ image_prefetcher_await_set() {
     fetcher_metrics="$(mktemp --suffix=.csv)"
     local fetcher_metrics_json
     fetcher_metrics_json="$(mktemp --suffix=.json)"
-    curl --silent --show-error --fail --retry 3 --retry-connrefused  "http://${endpoint}:8080/metrics" > "${fetcher_metrics_json}"
+    local metrics_url="http://${endpoint}:8080/metrics"
+    if ! curl --silent --show-error --fail --retry 3 --retry-connrefused "${metrics_url}" > "${fetcher_metrics_json}"; then
+        die "Failed to fetch prefetcher metrics from ${metrics_url}"
```
[the remark this answers is not in the document]
Comment (porridge) [reply scripts/ci/lib.sh:794]: Yes, that would be ideal, but without working metrics submission, we lose this channel :-(
Let's give it a chance. 🤞🏻
If we stumble upon a test job failure caused by a prefetcher failure, I'll be happy to remove the `die` calls and look for a different solution. 🫡

[the remark this answers is not in the document]
Comment (porridge) [reply scripts/ci/lib.sh:794]: Many thanks for reporting this @davdhacs , sorry it took me so long to look into this, things were rather hectic recently.
It seems this is related to AWS quota. Let's move the discussion to https://issues.redhat.com/browse/ROX-27359

**Pull request merged**